src/pdu.c

```diff
@@ -703,52 +737,125 @@ coap_pdu_parse_opt(coap_pdu_t *pdu) {
+               pdu->code >> 5, pdu->code & 0x1F,
+               (int)(opt_last - pdu->token - pdu->token_length));
+        good = 0;
+        break;
       }
       if (COAP_PDU_IS_SIGNALING(pdu)) {
         if (!coap_pdu_parse_opt_csm(pdu, len))
           goto bad;
-        continue;
       }
-      switch (pdu->max_opt) {
-      case COAP_OPTION_IF_MATCH:      if (len > 8) goto bad;              break;
-      case COAP_OPTION_URI_HOST:      if (len < 1 || len > 255) goto bad; break;
-      case COAP_OPTION_ETAG:          if (len < 1 || len > 8) goto bad;   break;
-      case COAP_OPTION_IF_NONE_MATCH: if (len != 0) goto bad;             break;
-      case COAP_OPTION_OBSERVE:       if (len > 3) goto bad;              break;
-      case COAP_OPTION_URI_PORT:      if (len > 2) goto bad;              break;
-      case COAP_OPTION_LOCATION_PATH: if (len > 255) goto bad;            break;
-      case COAP_OPTION_OSCORE:        if (len > 255) goto bad;            break;
-      case COAP_OPTION_URI_PATH:      if (len > 255) goto bad;            break;
-      case COAP_OPTION_CONTENT_FORMAT:if (len > 2) goto bad;              break;
-      case COAP_OPTION_MAXAGE:        if (len > 4) goto bad;              break;
-      case COAP_OPTION_URI_QUERY:     if (len < 1 || len > 255) goto bad; break;
-      case COAP_OPTION_HOP_LIMIT:     if (len < 1 || len > 1) goto bad;   break;
-      case COAP_OPTION_ACCEPT:        if (len > 2) goto bad;              break;
-      case COAP_OPTION_LOCATION_QUERY:if (len > 255) goto bad;            break;
-      case COAP_OPTION_BLOCK2:        if (len > 3) goto bad;              break;
-      case COAP_OPTION_BLOCK1:        if (len > 3) goto bad;              break;
-      case COAP_OPTION_SIZE2:         if (len > 4) goto bad;              break;
-      case COAP_OPTION_PROXY_URI:    if (len < 1 || len > 1034) goto bad; break;
-      case COAP_OPTION_PROXY_SCHEME:  if (len < 1 || len > 255) goto bad; break;
-      case COAP_OPTION_SIZE1:         if (len > 4) goto bad;              break;
-      case COAP_OPTION_NORESPONSE:    if (len > 1) goto bad;              break;
-      default:
-        ;
+      else if (!coap_pdu_parse_opt_base(pdu, len)) {
+        goto bad;
       }
       continue;
 bad:
-      coap_log(LOG_DEBUG,
-               "coap_pdu_parse: %d.%02d: option %u has bad length %u\n",
-               pdu->code >> 5, pdu->code & 0x1F, pdu->max_opt, len);
+      coap_log(LOG_WARNING,
+            "coap_pdu_parse: %d.%02d: offset %u option %u has bad length %u\n",
+               pdu->code >> 5, pdu->code & 0x1F,
+               (int)(opt_last - pdu->token - pdu->token_length), pdu->max_opt,
+               len);
       good = 0;
     }
 
+    if (!good) {
+      /*
+       * Dump the options in the PDU for analysis, space separated except
+       * error options which are prefixed by *
+       * Two rows - hex and ascii (if printable)
+       */
+      static char outbuf[100];
```

Comment:
Apparently, this buffer size does not fit for all options. I would like to have a pre-processor symbol to make it configurable. How about using `COAP_DEBUG_BUF_SIZE` for this as well, maybe similar to `coap_show_pdu()` in coap_debug.c`?

Reply:
Using COAP_DEBUG_BUF_SIZE instead of 100.